src/NzbDrone.Core/Update/InstallUpdateService.cs

```diff
@@ -153,7 +186,29 @@ private void EnsureAppDataSafety()
             if (_appFolderInfo.StartUpFolder.IsParentPath(_appFolderInfo.AppDataFolder) ||
                 _appFolderInfo.StartUpFolder.PathEquals(_appFolderInfo.AppDataFolder))
             {
-                throw new NotSupportedException("Update will cause AppData to be deleted, correct you configuration before proceeding");
+                throw new UpdateFailedException("Update will cause AppData to be deleted, correct you configuration before proceeding");
+            }
+        }
+
+        private void ExecuteInstallUpdate(Command message, UpdatePackage package)
+        {
+            try
+            {
+                InstallUpdate(package);
+
+                message.Completed("Restarting Sonarr to apply updates");
+            }
+            catch (UpdateFolderNotWritableException ex)
+            {
+                message.Failed(ex, string.Format("Startup folder not writable by user '{0}'", Environment.UserName));
+            }
+            catch (UpdateVerificationFailedException ex)
+            {
+                message.Failed(ex, "Update verification failed");
```

Comment:
It can't be too long coz it's shown on the UI notification. "Downloaded update package is corrupt"?